internal/sdkv2provider/resource_cloudflare_list_item.go

```diff
@@ -0,0 +1,217 @@
+	}
+
+	if list.Kind != listItemType {
+		return diag.FromErr(fmt.Errorf("items of type %s can not be added to lists of type %s", listItemType, list.Kind))
+	}
+
+	createListItemResponse, err := client.CreateListItem(ctx, cloudflare.AccountIdentifier(accountID), cloudflare.ListCreateItemParams{
+		ID:   listID,
+		Item: buildListItemCreateRequest(d),
+	})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("unable to create list item on list id %s: %w", listID, err))
+	}
+
+	newestItem := mostRecentlyCreatedItem(createListItemResponse)
+	d.SetId(newestItem.ID)
+
+	return nil
+}
+
+func resourceCloudflareListItemImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	attributes := strings.SplitN(d.Id(), "/", 3)
+
+	if len(attributes) != 3 {
+		return nil, fmt.Errorf("invalid id (\"%s\") specified, should be in format \"accountID/listID/itemID\"", d.Id())
+	}
+
+	accountID, listID, itemID := attributes[0], attributes[1], attributes[2]
+	d.SetId(itemID)
+	d.Set(consts.AccountIDSchemaKey, accountID)
+	d.Set("list_id", listID)
+
+	resourceCloudflareListItemRead(ctx, d, meta)
+
+	return []*schema.ResourceData{d}, nil
+}
+
+func resourceCloudflareListItemRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	accountID := d.Get(consts.AccountIDSchemaKey).(string)
+	listID := d.Get("list_id").(string)
+
+	listItem, err := client.GetListItem(ctx, cloudflare.AccountIdentifier(accountID), listID, d.Id())
+	if err != nil {
+		if strings.Contains(err.Error(), "could not find item") {
+			tflog.Info(ctx, fmt.Sprintf("List item %s no longer exists", d.Id()))
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(errors.Wrap(err, fmt.Sprintf("error reading List Item with ID %q", d.Id())))
+	}
+
+	d.Set("comment", listItem.Comment)
+
+	if listItem.IP != nil {
+		d.Set("ip", *listItem.IP)
+	}
+
+	if listItem.Redirect != nil {
+		optBoolToString := func(b *bool) string {
+			if b != nil {
+				switch *b {
+				case true:
+					return "enabled"
+				case false:
+					return "disabled"
+				}
+			}
+			return ""
+		}
+
+		d.Set("source_url", listItem.Redirect.SourceUrl)
+		d.Set("include_subdomains", optBoolToString(listItem.Redirect.IncludeSubdomains))
+		d.Set("target_url", listItem.Redirect.TargetUrl)
+		d.Set("status_code", listItem.Redirect.StatusCode)
+		d.Set("preserve_query_string", optBoolToString(listItem.Redirect.PreserveQueryString))
+		d.Set("subpath_matching", optBoolToString(listItem.Redirect.SubpathMatching))
+		d.Set("preserve_path_suffix", optBoolToString(listItem.Redirect.PreservePathSuffix))
+	}
+
+	return nil
+}
+
+func resourceCloudflareListItemUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// The API does not currently support in-place update of list items
+	dr := resourceCloudflareListItemDelete(ctx, d, meta)
```

Comment:
this is probably a bad idea because there will be a period of time where there will be no item in the list when folks are expecting it to be. instead, i would recommend we make the schema for `ip` and `redirect` to force new resources instead as that will defer the decision to the end user.